skipchain/struct.go

```diff
@@ -1027,3 +1112,57 @@ func (db *SkipBlockDB) getAllSkipchains() (map[string]*SkipBlock, error) {
 	}
 	return gen, nil
 }
+
+// skipBlockBuffer will cache a proposed block when the conode has
+// verified it and it will later store it in the DB after the protocol
+// has succeeded.
+// Blocks are stored per skipchain to prevent the cache to grow and it
+// is cleared after a new block is added (again per skipchain)
+type skipBlockBuffer struct {
+	buffer map[[32]byte]*SkipBlock
+	sync.Mutex
+}
+
+func newSkipBlockBuffer() *skipBlockBuffer {
+	return &skipBlockBuffer{
+		buffer: make(map[[32]byte]*SkipBlock),
+	}
+}
+
+// add stores the block using the skipchain ID as a key
+func (sbb *skipBlockBuffer) add(block *SkipBlock) {
+	sbb.Lock()
+	defer sbb.Unlock()
+
+	key := [32]byte{}
+	copy(key[:], block.SkipChainID())
+	sbb.buffer[key] = block
+}
+
+// get returns the block if the skipchain ID hits with the
+// correct block ID
+func (sbb *skipBlockBuffer) get(sid SkipBlockID, id SkipBlockID) (*SkipBlock, bool) {
```

Comment:
Why do you return a `bool` to indicate if the block is available or not? Isn't a `== nil` the same?